qlty-check/src/executor.rs

```diff
@@ -291,6 +292,14 @@ impl Executor {
             }
         }
 
+        let exported_config_files = self
+            .plan
+            .invocations
+            .iter()
+            .flat_map(|invocation| invocation.plugin.exported_config_files.clone())
+            .unique()
+            .collect_vec();
```

Comment:
To better align with our current separatio of concerns, it feels like the `Planner` should figure out exactly which exported config files are going to be used and which ones are going to be ignored (uniquely), and then encode that data into the `Plan` so that the Executor does not need to touch `plugin.exported_config_files`

Reply:
@brynary I'm a little confused by this. This approach intentionally follows the same one to [collect config_files](https://github.com/qltysh/qlty/blob/8bbc6da9016b756857dc82344c9f75e24a66f88b/qlty-check/src/executor.rs#L266) so as to be consistent. If we changed this, should we also change the above line? If not, why the difference?

Reply:
Ah, sorry for the confusion... I  think I would have the same observation about `collect_config_files` actually. However, I don't think we want to dig into that right now so let's leave this as is to confirm with the local code.